modules/hal_gigadevice/Kconfig

```diff
@@ -20,6 +20,34 @@ config HAS_GD32_HAL
 
 if HAS_GD32_HAL
 
+choice GD32_HXTAL_FREQUENCY
+	prompt "High speed external oscillator clock frequency"
+	default GD32_HXTAL_DEFAULT
+	help
+	  Define value of high speed crystal oscillator (HXTAL) in Hz
+	  This value sets the frequency of the oscillator.
+	  Only 8MHz and 25 MHz are supported.
+
+config GD32_HXTAL_DEFAULT
+	bool "Default"
```

Comment:
Not a good prompt.
This provides no valuable information to the user:
![image](https://user-images.githubusercontent.com/9802583/146165804-ef28acd5-6902-421f-a2bf-9ac157597887.png)

How about `Firmware defined`.

And then I believe that this setting should have some `depends on <SOC> / <BOARD>`.
Like
```
config GD32_HXTAL_FIRMWARE_DEFINED
	bool "Firmware defined"
        depends on <SOMETHING>
```

If you prefer not to have `depends on` knowledge at this level you can also place extra choice entries in a SoC or board specific Kconfig.

You can split a choice like this:
```
# main Kconfig file
choice MY_CHOICE
    prompt "my choices"

config MY_CHOICE_A
    bool "choice A"

endchoice
```

and then inside a board / SoC Kconfig you can add:
```
choice MY_CHOICE

config MY_CHOICE_EXTRA
    bool "This is an extra choice"

endchoice
```

Reply:
```
config GD32_HXTAL_FIRMWARE_DEFINED
```
It's good, I use this suggestion.

```
You can split a choice like this:
...
```

I'll use the first proposal.
8MHz and 25MHz are come from GigaDevice firmware.
I think it is preferable define in files under hal_gigadevice.
Should not define in another place.